File: 125-valid-palindrome/125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        if(s.isEmpty()){
            return true;
        }
        int start = 0;
        int end = s.length() - 1;
        while(start < end){
            //to check any non alpha-numeric and skip it
            while(start < end && !Character.isLetterOrDigit(s.charAt(start))){
                start++;
            }while(start < end && !Character.isLetterOrDigit(s.charAt(end))){
                end--;
            }
            if(Character.toLowerCase(s.charAt(start)) != Character.toLowerCase(s.charAt(end))){
                return false;
            }
            start++;
            end--;
        }
        return true;

    }
}

File: 125-valid-palindrome/125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    //base case
    if(s == null) return false;
    if(s.trim() === "") return true;
    //to convert into lower case
    let str = s.toLowerCase();
    let start = 0;
    let end = str.length - 1;
    while(start < end){
        while(start < end && !((str.charAt(start) >= 'a' && str.charAt(start) <= 'z') || (str.charAt(start) >= '0' && str.charAt(start) <= '9'))){
            start ++;
        }
        while(start < end && !((str.charAt(end) >= 'a' && str.charAt(end) <= 'z') || (str.charAt(end) >= '0' && str.charAt(end) <= '9'))){
            end--;
        }
        if(str.charAt(start) != str.charAt(end)){
            return false;
        }
        start++;
        end--;
        
    }
    return true;
};